src/setup_payload/AdditionalDataPayloadGenerator.h

```diff
@@ -0,0 +1,102 @@
+/**
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file provides a utility to generate Additional Data payload and its members
+ *      (e.g. rotating device id)
+ *
+ */
+
+#pragma once
+#include <core/CHIPError.h>
+#include <system/TLVPacketBufferBackingStore.h>
+#include <type_traits>
+
+namespace chip {
+namespace RotatingDeviceId {
+static constexpr unsigned kRotatingDeviceIdCounterStringMaxLength = 10;
+static constexpr unsigned kRotatingDeviceIdHashSuffixLength       = 16;
+static constexpr unsigned kRotatingDeviceIdMaxLength = kRotatingDeviceIdCounterStringMaxLength + kRotatingDeviceIdHashSuffixLength;
+static constexpr unsigned kRotatingDeviceIdHexMaxLength = kRotatingDeviceIdMaxLength * 2 + 1;
+} // namespace RotatingDeviceId
+
+enum class AdditionalDataFields : int8_t
+{
+    NotSpecified     = 0x00,
+    RotatingDeviceId = 0x01
+};
+
+inline AdditionalDataFields operator|(AdditionalDataFields lhs, AdditionalDataFields rhs)
```

Comment:
How about using `support/BitFlags.h` for setting/getting flags?

realistically the static_cast for the | values will not generate valid enum values and some users may find it surprising.

Reply:
Good point, i used BigFlags instead